mcs/class/Mono.Compiler/Mono.Compiler.BigStep/LLVM/BitCodeEmitter.cs

```diff
@@ -384,7 +390,42 @@ public void Process (OperationInfo opInfo)
                                 // tmp (array), tmp (index) => tmp
                                 InvokeOperation (op, exop, operands,
                                                         vm => {
-                                                                return GetIntArrayElement(operands[0].Type, 8, tempName, vm.Temp0, vm.Temp1);
+                                                                return GetIntArrayElement (operands[0].Type, 8, tempName, vm.Temp0, vm.Temp1);
+                                                        });
+                                break;
+                        case Opcode.StelemI1:
+                                // store element into array
+                                // tmp (array), tmp (index), tmp (value) => tmp
+                                InvokeOperation (op, exop, operands,
+                                                        vm => {
+                                                                SetIntArrayElement (operands[0].Type, 1, vm.Temp2, vm.Temp0, vm.Temp1);
+                                                        });
+                                break;
+                        case Opcode.StelemI2:
+                                // store element into array
+                                // tmp (array), tmp (index), tmp (value) => tmp
+                                InvokeOperation (op, exop, operands,
+                                                        vm => {
+                                                                SetIntArrayElement (operands[0].Type, 2, vm.Temp2, vm.Temp0, vm.Temp1);
+                                                        });
+                                break;
+                        case Opcode.StelemI:
+                                // ASSUME native integer = 4 bytes, may need change
```

Comment:
we only support AMD64 right now => 64 bit => native integer is 8 bytes. Anyway, that information should be received from some sort of "TargetConfiguration".